x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -140,7 +141,7 @@ public Realms(Settings settings, Environment env, Map<String, Realm.Factory> fac
 
     public List<Realm> asList() {
         final XPackLicenseState licenseStateSnapshot = licenseState.copyCurrentLicenseState();
-        if (licenseStateSnapshot.isSecurityEnabled() == false) {
+        if ( XPackSettings.SECURITY_ENABLED.get(settings) == false) {
             return Collections.emptyList();
```

Comment:
The two above are part of the same story that we don't have to check for Security enabled in all the trappy places to account for Security enabled status to change while something was in-progress.

I would remove them, but replacing these with asserts is an option too.

Reply:
:scissors:

Reply:
Do you think it's worth adding an `assert XPackSettings.SECURITY_ENABLED.get(settings) : "security must be enabled";` in all the components that should only be instantiated  by the security plugin ? 
`

Reply:
No, I think that passing the settings object around just to add asserts on its content is bad form, because the components are interdependent and the same assert would be evaluated multiple times, and we might find ourselves perpetuating this pattern to every new class we add.

I would use these only very sparingly, more to aid comprehension, to signal that we have a "logic chain" that tells Security must be on at this point so I don't have to look it up when I read the code.